newIDE/app/src/InAppTutorial/InAppTutorialOrchestrator.js

```diff
@@ -47,27 +97,179 @@ const translateAndInterpolateText = ({
+};
+
+const interpolateExpectedEditor = (
+  expectedEditor: {| editor: EditorIdentifier, scene?: string |} | null,
+  data: { [key: string]: string }
+): {| editor: EditorIdentifier, scene?: string |} | null => {
+  if (!expectedEditor) return null;
+  return {
+    ...expectedEditor,
+    scene: expectedEditor.scene ? data[expectedEditor.scene] : undefined,
+  };
+};
+
+const interpolateEditorTabActiveTrigger = (
+  trigger: string,
+  data: { [key: string]: string }
+): string => {
+  const [sceneKey, editorType] = trigger.split(':');
+  if (!editorType) {
+    throw new Error(`There might be missing a ":" in the trigger ${trigger}`);
+  }
+  if (editorType === 'Home') {
+    return `button[id="tab-start-page-button"][data-active="true"]`;
+  }
+  const sceneNameFilter = sceneKey ? `[data-scene="${data[sceneKey]}"]` : '';
+  return `button[id^="tab"][data-active="true"][data-type="${
+    editorType === 'Scene' ? 'layout' : 'layout-events'
+  }"]${sceneNameFilter}`;
+};
+
+export const getEditorTabSelector = ({
+  editor,
+  sceneName,
+}: {|
+  editor: EditorIdentifier,
+  sceneName?: string,
+|}): string => {
+  if (editor === 'Home') {
+    return 'button[id="tab-start-page-button"]';
+  }
+  const sceneNameFilter = sceneName ? `[data-scene="${sceneName}"]` : '';
+
+  return `button[id^="tab"][data-type="${
+    editor === 'Scene' ? 'layout' : 'layout-events'
+  }"]${sceneNameFilter}`;
+};
+
+const interpolateElementId = (
+  elementId: string,
+  data: { [key: string]: string }
+): string => {
+  if (
+    elementId.startsWith(selectorInterpolationProjectDataAccessors.editorTab)
+  ) {
+    const splittedElementId = elementId.split(':');
+    const sceneKey = splittedElementId[1];
+    // $FlowFixMe - We're confident the data is reliable
+    const editorType: EditorIdentifier = splittedElementId[2];
+    if (!editorType) {
+      throw new Error(
+        `There might be missing a ":" in the element id ${elementId}`
+      );
+    }
+    return getEditorTabSelector({
+      editor: editorType,
+      sceneName: data[sceneKey],
+    });
+  } else if (
+    elementId.startsWith(
+      selectorInterpolationProjectDataAccessors.objectInObjectsList
+    )
+  ) {
+    const splittedElementId = elementId.split(':');
+    const objectKey = splittedElementId[1];
+    return `#scene-editor[data-active] #objects-list div[data-object-name="${
+      data[objectKey]
+    }"]`;
+  } else if (
+    elementId.startsWith(
+      selectorInterpolationProjectDataAccessors.sceneInProjectManager
+    )
+  ) {
+    const splittedElementId = elementId.split(':');
+    const sceneKey = splittedElementId[1];
+    return `#project-manager [id^="scene-item"][data-scene="${
+      data[sceneKey]
+    }"]`;
+  } else if (
+    elementId.startsWith(
+      selectorInterpolationProjectDataAccessors.objectInObjectOrResourceSelector
+    )
+  ) {
+    const splittedElementId = elementId.split(':');
+    const objectKey = splittedElementId[1];
```

Comment:
you could probably have a function to get that. something like `getObjectKeyFromElementId()`